src/modules/avformat/common.c

```diff
@@ -20,9 +20,65 @@
 #include "common.h"
 
 #include <libavutil/channel_layout.h>
+#include <libavutil/pixdesc.h>
 #include <libavutil/samplefmt.h>
 
-int mlt_default_sws_flags = SWS_BICUBIC | SWS_FULL_CHR_H_INP | SWS_FULL_CHR_H_INT | SWS_ACCURATE_RND;
+int get_sws_flags(int srcwidth, int srcheight, int srcformat, int dstwidth, int dstheight, int dstformat)
+{
+	// Use default flags unless there is a reason to use something different.
+	int flags = SWS_BICUBIC | SWS_FULL_CHR_H_INP | SWS_FULL_CHR_H_INT | SWS_ACCURATE_RND;
+
+	if( srcwidth != dstwidth || srcheight != dstheight )
```

Comment:
This function is not currently used for any resolution scaling. But I added this so that it could be easily ported for use in filter_swscale. Also, this avoids the possibility that someone might accidentally use it for resolution scaling in the future and get unexpected results.